src/burn/engine/plan.cpp

```diff
@@ -2742,6 +2764,84 @@ static HRESULT FinalizeSlipstreamPatchActions(
     return hr;
 }
 
+static HRESULT PlanDependencyActions(
```

Comment:
A lot of this looks like it's from the changes I submitted a while back, but have since updated with bug fixes. Should probably take those as they have also been in production for quite a while. But you could rebase atop them with the support you added for ExePackage since I think we missed that.